src/main/java/minijava/parser/Parser.java

```diff
@@ -310,16 +311,10 @@ private void parseExpressionWithPrecedenceClimbing(int minPrecedence) {
 
   /** UnaryExpression -> PostfixExpression | (! | -) UnaryExpression */
```

Comment:
Could you add a test case that demonstrates the problem? Looking at the production, the existing code looks better than the new code ;) (Maybe the comment needs to change, too?)